public/js/navigator-js/integration/StateViewMap.js

```diff
@@ -43,33 +45,90 @@ this.navigatorjs.integration = this.navigatorjs.integration || {};
 				state = recipeStates[j];
 
 				if (requestedState.contains(state)) {
-					viewInstance = recipe.getViewInstance();
 
-					if (viewInstance.navigatorBehaviors instanceof Array) {
-						_addViewElementToDOM(recipe);
+					_addViewElementToDOM(recipe);
+					var viewInstance = recipe.getViewInstance();
+					if (Array.isArray(viewInstance.navigatorBehaviors)) {
 						_navigator.add(viewInstance, state);
 					}
+
+				} else {
+					if (recipe.isMounted()) {
+						_removeViewElementFromDOM(recipe);
+					}
 				}
 			}
 		}
 	}
 
+	function _removeViewElementFromDOM(recipe) {
```

Comment:
Side comment (i have not been following the discussion too closely) - I understand JSPerf is a good indicator of performance, but keep in mind that is over like hundreds of thousands (or millions) of operations on one function. When we may possibly only really going to be performing this a few times? 18% doesn't seem overly concerning to me.